server/apps/immich/src/api-v1/asset/asset.service.ts

```diff
@@ -44,10 +55,14 @@ export class AssetService {
     asset.modifiedAt = assetInfo.modifiedAt;
     asset.isFavorite = assetInfo.isFavorite;
     asset.mimeType = mimeType;
-    asset.duration = assetInfo.duration;
+    asset.duration = assetInfo.duration || null;
```

Comment:
OK, then the best place to fix it would be in the `AssetResponseDto`. So this line would correct, because that's just how it's saved in the DB. And I'll update the `AssetResponseDto` so `duration` is of type `string`, and when we map the entity to the DTO, if `duration` is null, it would return `0:00:00.00000`